lib/binary-parser.js

```diff
@@ -0,0 +1,95 @@
+export class BinaryParser {
+  constructor() {
+    this.schema = [];
+  }
+
+  string(fieldName, options) {
+    this.schema.push({type: 'string', fieldName, ...options});
+    return this;
+  }
+
+  int(fieldName, options) {
+    this.schema.push({type: 'int', fieldName, ...options});
+    return this;
+  }
+
+  buffer(fieldName, options) {
+    this.schema.push({type: 'buffer', fieldName, ...options});
+    return this;
+  }
+
+  array(fieldName, options) {
+    this.schema.push({type: 'array', fieldName, ...options});
+    return this;
+  }
+
+  parse(buffer, parentContext) {
+    const view = new DataView(buffer.buffer);
+    let offset = 0;
+    const context = {};
+    context.$parent = parentContext;
+    context.size = () => offset;
+
+    const readString = length => {
+      const value = new TextDecoder().decode(
+        buffer.slice(offset, offset + length)
```

Comment:
IIRC, this is faster because it won't copy the buffer when this is used in node.js (I *think* node.js's implementation copies the memory with `slice()` whereas the Web platform's does not -- at least when the `Uint8Array` instance also happens to be a node.js `Buffer`).

```suggestion
        buffer.subarray(offset, offset + length)
```